eZ/Publish/Core/FieldType/RichText/CustomTagsValidator.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\FieldType\RichText;
+
+use DOMDocument;
+
+/**
+ * Validator for Custom Tags input.
+ *
+ * The Validator checks if the given XML reflects proper Custom Tags configuration,
+ * mostly existence of specific Custom Tag and its required attributes.
+ */
+class CustomTagsValidator
+{
+    /**
+     * Custom Tags global configuration (ezpublish.ezrichtext.custom_tags Semantic Config).
+     *
+     * @var array
+     */
+    private $customTagsConfiguration;
+
+    /**
+     * @param array $customTagsConfiguration Injectable using %ezplatform.ezrichtext.custom_tags% DI Container parameter.
+     */
+    public function __construct(array $customTagsConfiguration)
+    {
+        $this->customTagsConfiguration = $customTagsConfiguration;
+    }
+
+    /**
+     * Validate Custom Tags found in the document.
+     *
+     * @param \DOMDocument $xmlDocument
+     *
+     * @return string[] an array of error messages
+     */
+    public function validateDocument(DOMDocument $xmlDocument)
```

Comment:
We don't translate those errors messages?

Reply:
No. Though I've seen somewhere in kernel package an attempt to translate errors, this is not a responsibility of a core package. API consumer is responsible for catching resulting exception and handling it any way he wants.